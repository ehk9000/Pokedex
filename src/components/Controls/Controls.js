import React from 'react';
import { Route } from 'react-router-dom';
import PokemonContainer from '../../containers/PokemonContainer/PokemonContainer';

export const Controls = () => {
  return (
    <section>
      <Route exact path='/' component={PokemonContainer} />
    </section>
  )
}

export default Controls;