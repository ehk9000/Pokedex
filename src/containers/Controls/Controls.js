import React from 'react';
import { Route } from 'react-router-dom';
import { connect } from 'react-redux';
import PokemonContainer from '../PokemonContainer/PokemonContainer';
import PokeDetails from '../../components/PokeDetails/PokeDetails';

export const Controls = (props) => {
  const pokemonList = props.pokemon
  return (
    <section>
      <Route exact path="/" component={PokemonContainer} />
      <Route 
        path="/pokemon/:id"
        render={({ match }) => {
          const { id } = match.params;
          const card = pokemonList.find(pokemon => pokemon.id === parseInt(id));
          if (card) {
            return <PokeDetails {...card} />
          }
        }}
      />
    </section>
  )
}

export const mapStateToProps = ({ pokemon }) => ({
  pokemon
});

export default connect(mapStateToProps)(Controls);