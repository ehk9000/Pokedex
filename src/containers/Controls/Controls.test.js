import React from 'react';
import { shallow } from 'enzyme';
import { Controls, mapStateToProps } from './Controls';

describe('Controls', () => {
  let wrapper;
  let mockPokemon = [{
    name: 'pikachu',
    type: 'electric',
    sprites: {
      front_default: 'something.png'
    },
    types: [
      {
        type: {
          name: 'electric'
        }
      }
    ],
    id: 25
  }]
  beforeEach(() => {
    wrapper = shallow( <Controls
        pokemon={mockPokemon} />)
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return a props object with pokemon', () => {
      const mockState = { pokemon: mockPokemon };
      const expected = { pokemon: mockPokemon };
      const mappedState = mapStateToProps(mockState);
  
      expect(mappedState).toEqual(expected);
    });
  });
});