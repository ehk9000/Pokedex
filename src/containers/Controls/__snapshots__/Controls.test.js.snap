// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls should render correctly 1`] = `ShallowWrapper {}`;
